streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -107,9 +107,6 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
    */
   def onStop()
 
-  /** Override this to specify a preferred location (hostname). */
-  def preferredLocation : Option[String] = None
```

Comment:
Since `Receiver` is an interface for user, I think it would be better not remove this method, one possible way you could is to pass the argument through `ReceiverInputDStream` to this buy adding some methods.